traits-stubs/traits-stubs/trait_types.pyi

```diff
@@ -605,12 +605,14 @@ class WeakRef(Instance):
         ...
 
 
-class Date(_BaseInstance[datetime.date]):
 
-    # simplified signature
+_OptionalDate = Optional[datetime.date]
```

Comment:
The other trait types don't seem to need this Optional bit. I don't think I understand why this is needed here.

Reply:
It's there because `None` is a valid value for this trait. The other trait types (`Datetime` and `Time`) *should* have this, I think, else mypy will complain about the assignment of `None`.

Reply:
(And arguably, `None` should not be a valid trait value here. That's where I'd like to get to, but we can't do it immediately because of backwards compatibility concerns.)

Reply:
Ah, it _is_ already there for `Datetime` and `Time`: the possibility of `None` is there in the `_BaseClass` generic definition. We've switched from using `_BaseInstance` as a base here to something more specific (and more accurate).